File: gems/decomplex/examples/kotlin/state-mesh.kt
class StateMeshExample {
  var a = 0
  var b = 0

  fun initialize() {
    this.a = 1
    this.b = 2
  }

  fun writer() {
    this.a = 3
  }

  fun reader(): Int {
    return this.a + this.b
  }

  fun a_alias(): Int {
    return this.a
  }
}

File: gems/decomplex/examples/ruby/state-mesh.rb
# frozen_string_literal: true

class StateMeshExample
  def initialize
    @a = 1
    @b = 2
  end

  def writer
    @a = 3
  end

  def reader
    @a + @b
  end

  def a_alias
    @a
  end
end
